src/main/java/vg/civcraft/mc/civmodcore/util/Iteration.java

```diff
@@ -169,4 +170,23 @@
 		return true;
 	}
 
+	/**
+	 * Determines whether a Map Entry is valid in that it exists and so does the key and value.
+	 *
+	 * @param entry The map entry itself.
+	 * @return Returns true if the entry is considered valid.
+	 */
+	public static boolean validEntry(Map.Entry<?, ?> entry) {
```

Comment:
This seems really unneccessary given that Entries should only be returned by iterating entry sets, which will never give a null Entry and that none of the common Map implementations (HashMap, TreeMap and EnumMap) allow null keys.

Isn't this just a null check on the value?

Reply:
Maybe I should've put into the javadoc that it's meant to be a stream filter rather than really anything else. It's akin to `.filter(Objects::nonNull)`